packages/caliper-core/test/worker/tx-observers/internal-tx-observer.js

```diff
@@ -0,0 +1,296 @@
+const sinon = require('sinon');
+const expect = chai.expect;
+const mockery = require('mockery');
+
+const warnLogger = sinon.stub();
+const errorLogger = sinon.stub();
+
+/**
+ * Mock implementation of CaliperUtils for testing purposes.
+ */
+class CaliperUtils {
+    /**
+     * Mocked version of the resolvePath method.
+     * Resolves the provided relative or absolute path and returns a fake path for testing.
+     *
+     * @param {string} path - The relative or absolute path to resolve.
+     * @returns {string} - Returns a hardcoded fake path, 'fake/path'.
+     */
+    static resolvePath(path) {
+        return 'fake/path';
+    }
+
+    /**
+     * Mocked version of the sleep method.
+     * Simulates a sleep by immediately resolving the promise.
+     *
+     * @param {number} ms - The number of milliseconds to sleep (ignored in the mock).
+     * @returns {Promise<void>} - A resolved promise, simulating the sleep.
+     */
+    static sleep(ms) {
+        return Promise.resolve();
+    }
+
+    /**
+     *
+     * @param {*} yaml res
+     * @return {string} the fake yaml
+     */
+    static parseYaml(yaml) {
+        return 'yaml';
+    }
+
+    /**
+     * Mocked version of the getLogger method.
+     * Returns a logger object with warn and error stubs for logging testing.
+     *
+     * @returns {{warn: function, error: function}} - A mocked logger with stubs for `warn` and `error` methods.
+     */
+    static getLogger() {
+        return {
+            warn: warnLogger,
+            error: errorLogger
+        };
+    }
+}
+
+
+mockery.enable({
+    warnOnReplace: false,
+    warnOnUnregistered: false
+});
+mockery.registerMock('../../common/utils/caliper-utils', CaliperUtils);
+
+describe('When using Internal Transaction Observer', () => {
+    const TxUpdateMessage = require('../../../lib/common/messages/txUpdateMessage');
+    const InternalTxObserver = require('../../../lib/worker/tx-observers/internal-tx-observer');
+    const TxResetMessage = require('../../../lib/common/messages/txResetMessage');
+    const ConfigUtil = require('../../../lib/common/config/config-util');
+
+
+    let messengerMock, managerUuid, workerIndex;
+    let clock;
+    const DEFAULT_UPDATE_INTERVAL = 1000; // in milliseconds
+
+    beforeEach(() => {
+        // Mocking dependencies
+        messengerMock = {
+            getUUID: sinon.stub().returns('mocked-messenger-uuid'),
+            send: sinon.stub().resolves()
+        };
+        managerUuid = 'mocked-manager-uuid';
+        workerIndex = 0;
+        clock = sinon.useFakeTimers();
+        warnLogger.reset();
+        errorLogger.reset();
+    });
+
+    afterEach(() => {
+        sinon.restore();
+        mockery.deregisterAll();
+        mockery.disable();
+        clock.restore();
+    });
+
+    describe('Construction', () => {
+        it('should initialize with correct properties', () => {
+            sinon.stub(ConfigUtil, 'get').returns(DEFAULT_UPDATE_INTERVAL);
```

Comment:
Now as DEFAULT_UPDATE_INTERVAL actually matches the real default, this line could be removed, however I think a better test would be
```
const TEST_UPDATE_INTERVAL=6000;
sinon.stub(ConfigUtil, 'get').withArgs(ConfigUtil.keys.Observer.Internal.Interval).returns(TEST_UPDATE_INTERVAL);
```